GUI.py

```diff
@@ -0,0 +1,647 @@
+        hbox.Add(cb, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.ALL, 1)
+        hbox.Add(self.chk_box, 0, wx.EXPAND | wx.ALIGN_CENTER_HORIZONTAL | wx.TOP, 5)
+
+        self.SetSizer(hbox)
+
+    def on_select(self, event: wx.Event) -> None:
+        i = event.GetString()
+        self.wvlngh, self.n_coef = refection_coef_read(os.path.join(self.cwd, i))
+
+    def on_open(self, event: wx.Event) -> None:
+
+        wildcard = "TXT files (*.txt)|*.txt"
+        dialog = wx.FileDialog(
+            self,
+            "Open Text Files",
+            wildcard=wildcard,
+            style=wx.FD_OPEN | wx.FD_FILE_MUST_EXIST,
+        )
+
+        if dialog.ShowModal() == wx.ID_CANCEL:
+            return
+
+        path = dialog.GetPath()
+
+        if os.path.exists(path):
+            self.data = np.loadtxt(path)
+
+        self.calc_btn.Enable(False)
+
+        draw_data(self.graph, self.data[:, 0], self.data[:, 1], name="Spectrum")
+
+        self.auto.Enable(True)
+
+        if self.graphs.IsShown():
+            self.graphs.Hide()
+            self.graph.Show()
+
+        self.Parent.Layout()
+        self.Parent.Fit()
+
+    def calc_choice(self, event: wx.Event) -> None:
+        """Executes calculation based on checked/unchecked box."""
+
+        if self.chk_box.GetValue():
+            self.make_report(*self.data_prep())
+        else:
+            self.ShowMessage(*self.data_prep())
+            while self.graph.toolbar.lx:
+                self.graph.toolbar.lx.pop(0).remove()
+            self.graph.canvas.draw()
+
+    def calc_auto(self, event: wx.Event) -> None:
+        """Returns thickness of the sample in a message box."""
+
+        self.graph.toolbar.x = [0, 9050.0, 9200.0]
+        self.ShowMessage(*self.data_prep(), auto_ga_as=True)
+        while self.graph.toolbar.lx:
+            self.graph.toolbar.lx.pop(0).remove()
+        self.graph.canvas.draw()
+
+    def ShowMessage(
+        self,
+        dat_X: Optional[np.ndarray] = None,
+        dat_Y: Optional[np.ndarray] = None,
+        wavelength: float = 9000.0,
+        n_wv_idx: Optional[float] = None,
+        n_true: float = 3.54,
+        auto_ga_as: bool = False,
+    ) -> None:
+        """Draws the message box with info about sample thickness."""
+
+        if auto_ga_as:
+            h = rolling_dist(dat_X=dat_X, dat_Y=dat_Y)
+        else:
+            self.psd, self.freq, true_freq, true_ind = fourier_analysis(dat_X, dat_Y)
+            h = thickness(wavelength, 1 / true_freq, n_true)
+
+        msg = "Sample thickness is {h} um".format(h=h)
+
+        dial = wx.MessageDialog(None, msg, "Info", wx.OK)
+        dial.ShowModal()
+        self.calc_btn.Enable(False)
+
+    def data_prep(self) -> Tuple[np.ndarray, np.ndarray, float, float, float]:
+        """Prepares data to be analysed."""
+        self.x = self.graph.toolbar.x[1:]
+        self.graph.toolbar.x[:] = []
+        self.x.sort()
+
+        try:
+            self.x != []
```

Comment:
You can check if it has length 0 or not 0.
+ Add some function for warnings I think, when user try to do smth weird.